File: CashFlow/Models/Request.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace CashFlow.Models;

public class Request
{
    public int Id { get; set; }
    public int UserId {get;set;}
    public int AccountId { get; set; }
    [Column(TypeName = "nvarchar(50)")]
    public RequestType Type { get; set; }
    //Normal cash
    public double AccountBalance { get; set; }
    public double AmountBalance { get; set; }
    public double FinallBalance { get; set; }
    //Credit
    public double AccountCredit { get; set; }
    public double AmountCredit { get; set; }
    public double FinallCredit { get; set; }
    
    public User? User { get; set; }
}

File: Frontend/cashflow/src/types/Request.ts
export class Request {
    id: number;
    userId: number;
    type: number;
    accountBalance: number;
    amountBalance: number;
    finallBalance: number;
    accountCredit: number;
    amountCredit: number;
    finallCredit: number;

    constructor(id: number, userId: number, type: number, accountBalance: number, amountBalance: number, finallBalance: number, accountCredit: number, amountCredit: number, finallCredit: number) {
        this.id = id;
        this.userId = userId;
        this.type = type;
        this.accountBalance = accountBalance;
        this.amountBalance = amountBalance;
        this.finallBalance = finallBalance;
        this.accountCredit = accountCredit;
        this.amountCredit = amountCredit;
        this.finallCredit = finallCredit;
    }
}
